pipelines/infrastructure/ol_infrastructure/instance_refresh_concourse_ami_ci.yaml

```diff
@@ -0,0 +1,98 @@
+resource_types:
+- name: ami
+  type: docker-image
+  source:
+    repository: jdub/ami-resource
+- name: pulumi
+  source:
+    repository: mitodl/concourse-pulumi-resource-provisioner
+  type: docker-image
+
+resources:
+- name: concourse-worker-ami
+  type: ami
+  check_every: 1m
+  source:
+    region: us-east-1
+    filters:
+      owner-id: 610119931565
+      is-public: false
+      state: available
+      name: concourse-worker-*
+
+- name: concourse-web-ami
+  type: ami
+  check_every: 1m
+  source:
+    region: us-east-1
+    filters:
+      owner-id: 610119931565
+      is-public: false
+      state: available
+      name: concourse-web-*
+
+- icon: github
+  name: pulumi_files
+  source:
+    branch: main
+    paths:
+    - src/ol_infrastructure/infrastructure/concourse
+    uri: https://github.com/mitodl/ol-infrastructure
+  type: git
+
+- name: pulumi-infra
+  source: null
+  type: pulumi
+
+jobs:
+- name: concourse-ami-build
+  plan:
+  - task: set-aws-creds
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: amazon/aws-cli
+          tag: 'latest'
+      params:
+        AWS_ROLE: arn:aws:iam::610119931565:role/ol-applications/concourse/role/concourse-instance-role-worker-infra-qa-4c8b313
```

Comment:
Do we need to hard-code this? Do we need to pass it in at all? It would be great if we can pull it dynamically from the metadata API via curl or something similar.